cores/esp8266/reboot_uart_dwnld.cpp

```diff
@@ -0,0 +1,112 @@
+#include "reboot_uart_dwnld.h"
+#include <stdnoreturn.h>
+#include <ets_sys.h>
+#include <user_interface.h>
+#include <spi_flash.h>
+#include <esp8266_undocumented.h>
+
+void ICACHE_RAM_ATTR boot_from_something_uart_dwnld(void (**user_start_ptr)())
+{
+	/* simplified for following condition
+	 * (GPI >> 0x10 & 0x07) == 1 and
+	 * (GPI >> 0x1D & 0x07) == 6
+	 */
+
+	const uint32_t uart_no = 0;
+	const uint16_t divlatch = uart_baudrate_detect(uart_no, 0);
+	uart_div_modify(uart_no, divlatch);
+	UartDwnLdProc((uint8_t*)0x3fffa000, 0x2000, user_start_ptr);
+}
+
+[[noreturn]] void ICACHE_RAM_ATTR main_uart_dwnld()
+{
+	// TODO may be it is part of _start()
+	// at least ets_intr_lock() is called in system_restart_local()
+	// Therefore unlock it here again to enable UART Rx IRQ
+	ets_intr_unlock();
+
+	uartAttach();
+	Uart_Init(0);
+	ets_install_uart_printf(0);
+
+	boot_from_something_uart_dwnld(&user_start_fptr);
+
+	// Call _ResetVector() instead of original implementation
+	// Therefore esptool.py stub is not yet supported
+	/* moved from system_restart_core_uart_dwnld() to here
+	 * Without this bit clearing the _ResetVector() would run into
+	 * an exception
+	 */
+	CLEAR_PERI_REG_MASK(PERIPHS_DPORT_24, 0x18);
+	_ResetVector();
+
+	// TODO calling user_start_fptr() fails with an exception
```

Comment:
I don't quite understand the purpose of the comment or commented out code.  Remove?

Reply:
Previously it was not working with esptool.py flasher stub (--no-stub) option was required. The commented part was at least required to support esptool.py flasher stub.

I have solved the issue and esptool.py flasher stub mode is supported, now.